File: 3.1 - Clock Class /Clock/Clock/Clock/Counter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace Clock
{
    public class Counter
    {

        private int _count;
        private string _name;

        public Counter(string name)
        {
            _count = 0;
            _name = name;
        }

        public void Reset()
        {
            _count = 0;
        }

        public void Increment()
        {
            _count++;
        }

        public string Name
        {
            get
            {
                return _name;
            }
            set
            {
                _name = value;
            }
        }

        public int Tick
        {
            get
            {
                return _count;
            }
        }
    }
}

File: 11.1 - Clock in Another Language/Python/Counter.py
class Counter:
    def __init__(self, name):
        self._count = 0
        self._name = name
    
    def Increment(self):
        self._count += 1
        
    def Reset(self):
        self._count = 0
        
    @property
    def Name(self):
        return self._name
    
    @Name.setter
    def Name(self, value):
        self._name = value

    @property 
    def Ticks(self):
        return self._count
